ddtrace/contrib/django/patch.py

```diff
@@ -466,7 +467,7 @@ def traced_get_response(django, pin, func, instance, args, kwargs):
 @trace_utils.with_traced_module
 def traced_template_render(django, pin, wrapped, instance, args, kwargs):
     """Instrument django.template.base.Template.render for tracing template rendering."""
-    template_name = getattr(instance, "name", None)
+    template_name = maybe_stringify(getattr(instance, "name", None))
```

Comment:
Given that a template can be a template object, a path to a template, or a list of paths, I would rather avoid such coercion if we could.

https://github.com/django/django/blob/1710cdbe79c90665046034fe1700933d038d90ad/django/template/response.py#L60-L67
https://github.com/django/django/blob/1710cdbe79c90665046034fe1700933d038d90ad/django/template/loader.py#L22-L50

Reply:
Isn't `instance` guaranteed to be an instance of `Template` in this case, given that we are in the wrapper for `Template.render` here? Then perhaps it's relatively safe to assume that the attribute `name`, whenever set, could be converted to string. WDYT?

Reply:
Sorry I had meant to reply to this earlier. I see what you mean. It appears the `name` property on the instance is not always a `str`, though as you point out, should be coerced to a string. I'd still would like to have coverage of the different types we expect to encounter. If it's just `PosixPath` (which implements `__str__`) then we can be sure the resulting span data is as expected. Is there another template name we can come across? What would it's `resource` and span tags look like? Perhaps we can get this for free from the Django test suite but we'd need to be able to capture such unexpected behavior.

Reply:
@majorgreys I've added a test case for the tag and the resource attribute when the template name is an instance of `PosixPath`.